cupy/statistics/histogram.py

```diff
@@ -217,29 +218,25 @@ def histogram(x, bins=10, range=None, weights=None, density=False):
                     and x.size <= 0x7fffffff and bin_edges.size <= 0x7fffffff):
                 # Need to ensure the dtype of bin_edges as it's needed for both
                 # the CUB call and the correction later
-                if isinstance(bins, cupy.ndarray):
-                    bin_type = cupy.result_type(bin_edges, x)
-                    if cupy.issubdtype(bin_type, cupy.integer):
-                        bin_type = cupy.result_type(bin_type, float)
-                    bin_edges = bin_edges.astype(bin_type, copy=False)
+                assert isinstance(bin_edges, cupy.ndarray)
+                bin_type = numpy.result_type(bin_edges.dtype, x.dtype)
+                if numpy.issubdtype(x.dtype, numpy.integer):
+                    bin_type = numpy.float
+                else:
+                    # TODO(okuta): support float16
+                    if (bin_type == numpy.float16 and
+                            int(device.get_compute_capability()) < 53):
+                        bin_type = numpy.float32
+                    x = x.astype(bin_type, copy=False)
```

Comment:
Is this for `x.dtype == numpy.float16`? If so, it should be indented following the line above. Otherwise, I don't see why we need a type conversion here 😅

Reply:
I will add a corresponding test case.